diff --git a/src/input/pointer.c b/src/input/pointer.c
index 630e7d6..e6c9c1b 100644
--- a/src/input/pointer.c
+++ b/src/input/pointer.c
@@ -983,12 +983,10 @@ void pointer_process_motion(uint32_t time, struct wlr_input_device *device,
 		wlr_idle_notifier_v1_notify_activity(server.idle_notifier, server.seat);

 		/* Update selected_monitor (even while dragging a window) */
-		if (config.sloppyfocus) {
-			Monitor *oldmon = server.selected_monitor;
-			set_selected_monitor(
-				monitor_at_point(server.cursor->x, server.cursor->y));
-			if (oldmon != server.selected_monitor)
-				printstatus(IPC_WATCH_MONITOR | IPC_WATCH_ALL_MONITORS);
-		}
+		Monitor *newmon = monitor_at_point(server.cursor->x, server.cursor->y);
+		if (newmon && newmon != server.selected_monitor) {
+			set_selected_monitor(newmon);
+			printstatus(IPC_WATCH_MONITOR | IPC_WATCH_ALL_MONITORS);
+		}
 	}
